src/audio/module_adapter/module/iadk_modules.c

```diff
@@ -102,7 +130,14 @@ static int iadk_modules_init(struct processing_module *mod)
 	md->mpd.out_buff_size = src_cfg->obs;
 
 	/* Call module specific init function if exists. */
-	ret = iadk_wrapper_init(mod->priv.module_adapter);
+	if (mod->is_loadable) {
```

Comment:
it is also quite confusing to use the iadk_module_init() and this whole file in general for non-IADK modules as well. If it is generic, maybe we need to start with renaming the file and the functions first